pkg/sanity/sanity.go

```diff
@@ -32,6 +32,24 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+// Valid CSI driver roles, used for parsing tests
+type DriverRole string
+type DriverRoles []DriverRole
+
+func (drs DriverRoles) String() []string {
+	strs := []string{}
+	for _, d := range drs {
+		strs = append(strs, string(d))
+	}
+	return strs
+}
+
+const (
+	DriverRoleIdentity   DriverRole = "Identity"
+	DriverRoleController DriverRole = "Controller"
+	DriverRoleNode       DriverRole = "Node"
```

Comment:
It is not obvious that these strings are sub-strings of the first parameter of each `DescribeSanity` call. Perhaps this should be made explicit and enforced by turning the first parameter of that function into a `DriverRole` and then appending the common ` Service` suffix in the implementation of that call.

Reply:
I was explicitly trying to avoid this, because I didn't want to add a new parameter to the test functions nor go and update every single existing testing function to a new format. I can add a comment above them to describe this relationship, if that would help?

Reply:
Such a comment would have to be added to `sanity.go` and all calls to `DescribeSanity`, because otherwise someone looking at those calls won't know that the strings have a special meaning. Then one might as well change the call.